tests/src/test_account.cpp

```diff
@@ -0,0 +1,38 @@
+#include "es-test.hpp"
+
+// Include only the tested class's header.
+#include "../../source/Account.h"
+
+// ... and any system includes needed for the test file.
+#include <map>
+#include <string>
+
+
+
+namespace { // test namespace
+
+// #region mock data
+// #endregion mock data
+
+
+
+// #region unit tests
+SCENARIO( "Creating an Account" , "[Account][Creation]" ) {
+	GIVEN( "initial values" ) {
+		Account account;
+		THEN( "storing and retrieving data should work" ) {
+			REQUIRE( account.Credits() == 0. );
+			account.AddCredits(100);
```

Comment:
This would align better with bdd as e.g.

```
Given an account
When money is added
Then the balance is increased
```
And
```
Given an account
When a fine is levied
Then debt is incurred
```

So the code would have a root `given` + account definition, then sibling `when`s that `require` a certain balance / debt (since their follow-up `then` won't be valid if the requires fail) and perform the modification, and lastly `then`s whose `check`s verify the balance / debt exist and are correct.